src/esssans/beam_center_finder.py

```diff
@@ -381,6 +404,15 @@ def beam_center_from_iofq(
     logger.info(f'Using minimizer: {minimizer}')
     logger.info(f'Using tolerance: {tolerance}')
 
+    # Flatten positions dim which is required during the iterations for slicing with a
+    # boolean mask
+    pos_dims = data.coords['position'].dims
+    new_dim = uuid.uuid4().hex
+    data = data.flatten(dims=pos_dims, to=new_dim)
+    dims_to_flatten = [dim for dim in norm.dims if dim in pos_dims]
+    if dims_to_flatten:
+        norm = norm.flatten(dims=dims_to_flatten, to=new_dim)
+
```

Comment:
If you are refering to the flattening inside `beam_center_from_center_of_mass`, then yes flattening is still required here because when selecting quadrants defined by a boolean condition on phi, we need the data to be 1d.
You want to flatten here because you want to do it only once, not at every iteration (it would have made more sense to keep the flattening closer to where it is actually needed, i.e. inside `_iofq_in_quadrants`, but it would then be more expensive).

Now you could ask that if we flatten here, maybe we don't need to flatten above? However, we also call `beam_center_from_center_of_mass` directly, without going via `beam_center_from_iofq`, so we also need to flatten there. It just means that if we go via `beam_center_from_iofq`, the flattening in `beam_center_from_center_of_mass` won't have any effect.